Sources/ClientGtk/ServerListView.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftCrossUI
+import DeltaCore
+
+enum DetailState {
+  case addition, server(ServerDescriptor), error(String)
+}
+
+class ServerListViewState: Observable {
+  @Observed var servers: [ServerDescriptor] = []
+  @Observed var detailState = DetailState.addition
+}
+
+struct ServerListView: View {
+  var completionHandler: (ServerDescriptor) -> Void
+
+  var state = ServerListViewState()
+
+  var body: some ViewContent {
+    NavigationSplitView {
+      VStack {
+        Button("Add server") {
+          state.detailState = .addition
+        }
+
+        ScrollView {
+          ForEach(state.servers) { server in
+            Button(server.name) {
+              state.detailState = .server(server)
+            }
+            .padding(Edge.Set.bottom, 5)
+          }
+        }
+      }
+      .frame(minWidth: 100)
+      .padding(Edge.Set.trailing, 10)
+    } detail: {
+      VStack {
+        switch state.detailState {
+          case .addition:
+            additionView
+          case .server(let server):
+            Text(server.name)
+            if let port = server.port {
+              Text(server.host + ":" + String(port))
+            } else {
+              Text(server.host)
+            }
+
+            Button("Connect") {
+              completionHandler(server)
+            }
+          case .error(let message):
+            Text(message)
+        }
+      }
+      .padding(Edge.Set.leading, 10)
+    }
+  }
+
+  var additionView: some View {
+    VStack {
+      @Observed var name = ""
```

Comment:
State variables here won't get tracked at all (same is true in SwiftUI), so it'd be very easy to run into weird behaviour by doing this. If a parent view's state changed while the user was inputting stuff, the inputs would just reset. Instead add `name` and `address` to the view's state.